### DISCUSSION TITLE
Questions on settings.json and loading models, modes, and extensions...

### DISCUSSION THREAD

**Discussion by juicyroots** [oobabooga/text-generation-webui#1947] [Q&A]
I have my `settings.json` file in the root and launching with `python server.py --listen --api --auto-devices --settings settings.json` in my `webui.py`, but I can't seem to get it to load in chat mode, load a model, or extensions. The command line says, `INFO:Loading settings from settings.json...`, and if I change simple values (such as truncation_length or chat_prompt_size values), they are reflected the UI. So I know part of it is working, just not models, modes, and extensions. Model is always none, Mode is always default, and extensions are never checked. 

I searched everywhere for how this works or an example setup (other than the example in the repo), but I'm at a loss. Can someone help me understand what I'm missing? Any help would be appreciated.

```
{
    "model": "PygmalionAI_pygmalion-6b",
	"max_new_tokens": 200,
    "max_new_tokens_min": 1,
    "max_new_tokens_max": 2000,
    "seed": -1,
    "character": "None",
    "name1": "You",
    "name2": "Assistant",
    "context": "This is a conversation with your Assistant. The Assistant is very helpful and is eager to chat with you and answer your questions.",
    "greeting": "",
    "turn_template": "",
    "custom_stopping_strings": "",
    "stop_at_newline": false,
    "add_bos_token": true,
    "ban_eos_token": false,
    "skip_special_tokens": true,
    "truncation_length": 2048,
    "truncation_length_min": 0,
    "truncation_length_max": 8192,
    "mode": "chat",
    "chat_style": "cai-chat",
    "instruction_template": "None",
    "chat_prompt_size": 2048,
    "chat_prompt_size_min": 0,
    "chat_prompt_size_max": 2048,
    "chat_generation_attempts": 1,
    "chat_generation_attempts_min": 1,
    "chat_generation_attempts_max": 5,
    "default_extensions": [],
    "chat_default_extensions": [
        "gallery"
    ],
    "presets": {
        "default": "Default",
        ".*(alpaca|llama|llava)": "LLaMA-Precise",
        ".*pygmalion": "NovelAI-Storywriter",
        ".*RWKV": "Naive"
    },
    "prompts": {
        "default": "QA",
        ".*(gpt4chan|gpt-4chan|4chan)": "GPT-4chan",
        ".*oasst": "Open Assistant",
        ".*alpaca": "Alpaca"
    },
    "lora_prompts": {
        "default": "QA",
        ".*(alpaca-lora-7b|alpaca-lora-13b|alpaca-lora-30b)": "Alpaca"
    }
}
```

**Comment by kha84** [comment]
I think a lot of code has changed since the moment you asked this question. Have you tried the most recent versions?
I can guess, you might need to push the "Save settings" button in UI, to reflect all the config changes you've made to be saved to settings.yaml.

I was having some small issues on my own, not really understanding, how to prevent WebUI from auto-loading the very first ggml model it finds, if nothing is specified on settings. Because I was having issues with llama-cpp-python package, the WebUI was crashing all the time, while trying to load one and only ggml model I had. And I found no way of bypassing it, rather than to temporary move the model out from models/ directory. Changing the model name to empty in settings.yaml wasn't helping.
